src/GraphUtils/utils.js

```diff
@@ -1,3 +1,29 @@
+/**
+ * Get subgroup links from link
+ * @param {array} link - array of links
+ * @param {object} nameToNode - key (node name) value (node object) map
+ * @param {string} sourceId - source id for subgroup links
+ * This function traverse links recursively and return all nested subgroup lnks
+ */
+const getSubgroupLinks = (link, nameToNode, sourceId) => {
+  let subgroupLinks = [];
+  if (link.subgroup) {
+    link.subgroup.forEach((sgLink) => {
+      if (sgLink.subgroup) {
+        subgroupLinks = subgroupLinks.concat(getSubgroupLinks(sgLink, nameToNode, sourceId));
+      } else {
+        subgroupLinks.push({
+          source: nameToNode[sourceId],
+          target: nameToNode[sgLink.target_type],
+          exists: 1,
```

Comment:
what is "exists" for?

Reply:
that's for previous logic in `createNodesAndEdges`, I think it's used for filtering out 0 count links for data model graph.